### PR TITLE
feat: add gccollab module

### PR DESCRIPTION
**Pull request by mfan127**

### CODE REVIEW

File: src/Modules/StatCan.OrchardCore.GCCollab/Drivers/GCCollabAuthenticationSettingsDisplayDriver.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                {
+                    return null;
+                }
+
+                var model = new GCCollabAuthenticationSettingsViewModel();
+                await context.Updater.TryUpdateModelAsync(model, Prefix);
+
+                if (context.Updater.ModelState.IsValid)
+                {
+                    var protector = _dataProtectionProvider.CreateProtector(GCCollabConstants.Features.GCCollabAuthentication);
```
Comment (jptissot) [inline src/Modules/StatCan.OrchardCore.GCCollab/Drivers/GCCollabAuthenticationSettingsDisplayDriver.cs:80]: Move protector inside if below

File: src/Modules/StatCan.OrchardCore.GCCollab/Drivers/GCCollabAuthenticationSettingsDisplayDriver.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+                    settings.ClientID = model.ClientID;
+                    if(!(model.ClientSecret == MagicString))
```
Comment (jptissot) [inline src/Modules/StatCan.OrchardCore.GCCollab/Drivers/GCCollabAuthenticationSettingsDisplayDriver.cs:83]: model.ClientSecret != MagicString

Comment (jptissot) [reply src/Modules/StatCan.OrchardCore.GCCollab/Drivers/GCCollabAuthenticationSettingsDisplayDriver.cs:83]: Maybe a blank check as well, unless the protector already handles it.

Comment (mfan127) [reply src/Modules/StatCan.OrchardCore.GCCollab/Drivers/GCCollabAuthenticationSettingsDisplayDriver.cs]: Not sure if data protector handles it or not I just checked if it's empty.

**Pull request merged**